src/Hosting/Hosting/test/HostingApplicationDiagnosticsTests.cs

```diff
@@ -535,10 +535,39 @@ public void ActivityBaggageReadFromLegacyHeaders()
         features.Set<IHttpRequestFeature>(new HttpRequestFeature()
         {
             Headers = new HeaderDictionary()
+            {
+                {"baggage", "Key1=value1, Key2=value2"}
```

Comment:
Not having a request id here exercises the new code (thanks!), but is it something that could plausibly happen?  This question may be terribly naive - I'm not very familiar with distributed tracing.

Reply:
Yes it's possible. Not all frameworks set it. This is not the case with .NET because `HttpClient` sets it (https://github.com/dotnet/runtime/blob/33a68d0cef067034df5cbac89415568e5fda63ce/src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs#L308) but someone can use another framework (or make the request manually via curl, Postman, or whatever) and in this case there will be no request id (`traceparent`) in the request (ASP .NET will generate one in this case).